osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -178,32 +174,18 @@ private async Task<ServerMultiplayerRoom> retrieveRoom(long roomId)
                 if (databaseRoom.user_id != CurrentContextUserId)
                     throw new InvalidStateException("Non-host is attempting to join match before host");
 
-                var playlistItem = await db.GetCurrentPlaylistItemAsync(roomId);
-                var beatmapChecksum = await db.GetBeatmapChecksumAsync(playlistItem.beatmap_id);
-
-                if (beatmapChecksum == null)
-                    throw new InvalidOperationException($"Expected non-null checksum on beatmap ID {playlistItem.beatmap_id}");
-
-                var requiredMods = JsonConvert.DeserializeObject<IEnumerable<APIMod>>(playlistItem.required_mods ?? string.Empty) ?? Array.Empty<APIMod>();
-                var allowedMods = JsonConvert.DeserializeObject<IEnumerable<APIMod>>(playlistItem.allowed_mods ?? string.Empty) ?? Array.Empty<APIMod>();
-
-                var room = new ServerMultiplayerRoom(roomId, this)
+                var room = new ServerMultiplayerRoom(roomId, databaseFactory, this)
```

Comment:
I know we agreed on passing in the `databaseFactory` here, but for the `Initialise()` calls specifically, I'd feel a lot happier with them if they were `Initialise(databaseFactory)`, implying that the `ctor` didn't do the actual room/database setup via the method structure.

Doesn't need to change, just mentioning in passing.

Reply:
So passing database factory into the ctor but also passing it into the initialise? Or are you expecting initialise to store the factory?

Reply:
Wasn't expecting any change. But what you mention would be better if it can work, I think?

Reply:
Moved `IDatabaseFactory` into `Initialise()`, see if you agree: https://github.com/ppy/osu-server-spectator/pull/84/commits/bf5372a77ea027d8f47dc1e696b14bde2b1418dd